client/dashboard/sites/settings-repositories/connect-repository-form.tsx

```diff
@@ -0,0 +1,437 @@
+import {
+	githubInstallationsQuery,
+	githubRepositoriesQuery,
+	githubRepositoryBranchesQuery,
+	githubRepositoryChecksQuery,
+} from '@automattic/api-queries';
+import { useQuery } from '@tanstack/react-query';
+import {
+	Button,
+	SelectControl,
+	ToggleControl,
+	ComboboxControl,
+	TextControl,
+	RadioControl,
+	__experimentalVStack as VStack,
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { AdvancedWorkflowValidation } from './advanced-workflow-validation';
+import { useCreateCodeDeployment } from './use-create-code-deployment';
+import type { Site, GitHubInstallation, GitHubRepository } from '@automattic/api-core';
+
+interface ConnectRepositoryFormProps {
+	site: Site;
+	onConnected: () => void;
+	onCancel: () => void;
+}
+
+const INSTALL_APP_URL = 'https://github.com/apps/wordpress-com/installations/new';
+
+export const ConnectRepositoryForm = ( {
+	site,
+	onConnected,
+	onCancel,
+}: ConnectRepositoryFormProps ) => {
+	const [ selectedInstallationId, setSelectedInstallationId ] = useState< number | '' >( '' );
+	const [ selectedRepositoryId, setSelectedRepositoryId ] = useState< number | '' >( '' );
+	const [ branch, setBranch ] = useState( '' );
+	const [ targetDir, setTargetDir ] = useState( '/' );
+	const [ isAutomated, setIsAutomated ] = useState( false );
+	const [ isSubmitting, setIsSubmitting ] = useState( false );
+	const [ isTargetDirDirty, setIsTargetDirDirty ] = useState( false );
+	const [ deploymentMode, setDeploymentMode ] = useState< 'simple' | 'advanced' >( 'simple' );
+	const [ workflowPath, setWorkflowPath ] = useState< string | undefined >( undefined );
+
+	const {
+		data: installations = [],
+		isLoading: isLoadingInstallations,
+		error: installationsError,
+		refetch: refetchInstallations,
+	} = useQuery( githubInstallationsQuery() );
```

Comment:
I think there's too many `useEffect()`s in this file that can be simplified if we preload `githubInstallationsQuery()` in the router, and then use `useSuspenseQuery()` here. This way `installations` will be always defined. What do you think 🤔

Reply:
Oh I see what you mean. Got it, feel free to do the simplest way for you for now :)